### PR TITLE
[SRK-11] Add more HLint rules

### PR DESCRIPTION
**Pull request by neongreen**

### CODE REVIEW

File: HLint.hs [hunk 1]
```diff
@@ -75,6 +77,16 @@ warn "Avoid 'both'" = both ==> Control.Lens.each
     \  * If you want to traverse all elements of the tuple, use 'each'.\n\
     \  * If 'both' is used on 'Either' here, replace it with 'chosen'."
 
+warn = either (const True) (const False) ==> isLeft
+warn = either (const False) (const True) ==> isRight
+
+warn = Data.Map.toAscList (Data.Map.fromList x) ==> Universum.sortOn fst x
+warn = Data.Map.toDescList (Data.Map.fromList x) ==> Universum.sortOn (Down . fst) x
+
+warn = forM_ ==> for_
```
Comment (volhovM) [inline HLint.hs]: Motivation?

File: HLint.hs [hunk 2]
```diff
@@ -139,22 +151,271 @@ suggest = or <$> mapM f s ==> Universum.anyM f s
 --     suggest = (do x <- m; when x a) ==> Universum.whenM m a
 --     suggest = (do x <- m; unless x a) ==> Universum.unlessM m a
 
-warn = (case m of Just x -> f x; Nothing -> pure ()) ==> Universum.whenJust m f
-warn = (case m of Just x -> f x; Nothing -> return ()) ==> Universum.whenJust m f
+warn = whenM (not <$> x) ==> unlessM x
+warn = unlessM (not <$> x) ==> whenM x
 
-warn = (case m of Nothing -> pure (); Just x -> f x) ==> Universum.whenJust m f
+-- Oh boy, we sure have many ways of spelling “pure ()”. Also I checked and
```
Comment (volhovM) [inline HLint.hs]: I really hope next 200 lines of diff are impossible to make more succinct. 
:(

File: src/Pos/Txp/DB/Balances.hs [hunk 3]
```diff
@@ -108,14 +108,10 @@ prepareGStateBalances
...
-    putIfEmpty getRealStakeSumMaybe putFtsStakes
-    putIfEmpty getRealStakeSumMaybe putGenesisTotalStake
+    whenNothingM_ getRealStakeSumMaybe putFtsStakes
+    whenNothingM_ getRealStakeSumMaybe putGenesisTotalStake
   where
     totalCoins = sumCoins $ map snd $ concatMap txOutStake $ toList genesisUtxo
-    putIfEmpty
-        :: forall a.
-           (m (Maybe a)) -> m () -> m ()
-    putIfEmpty getter putter = maybe putter (const pass) =<< getter
```
Comment (volhovM) [inline src/Pos/Txp/DB/Balances.hs]: Hilarious.

File: src/Pos/Types/Arbitrary.hs [hunk 4]
```diff
@@ -24,7 +24,6 @@ module Pos.Types.Arbitrary
 import           Universum
 
 import qualified Data.ByteString            as BS (pack)
-import           Data.Char                  (chr)
```
Comment (volhovM) [inline src/Pos/Types/Arbitrary.hs]: Why didn't we have warning for this one? 🤔

File: HLint.hs (hunk 1 above)
Comment (neongreen) [inline HLint.hs]: Consistency (whenever I see `forM_` it looks weird)

**Pull request merged**